app/controllers/api/v1/display_ads_controller.rb

```diff
@@ -15,11 +15,40 @@ def show
         render json: @display_ad
       end
 
+      def create
+        @display_ad = DisplayAd.new(permitted_params)
+        result = @display_ad.save
+        render json: @display_ad, status: (result ? :ok : :unprocessable_entity)
+      rescue ArgumentError => e
+        # enums raise ArgumentError exceptions on unexpected inputs!
+        render json: { error: e }, status: :unprocessable_entity
+      end
+
+      def update
+        @display_ad = DisplayAd.find(params[:id])
+        result = @display_ad.update(permitted_params)
+        render json: @display_ad, status: (result ? :ok : :unprocessable_entity)
+      rescue ArgumentError => e
+        # enums raise ArgumentError exceptions on unexpected inputs!
+        render json: { error: e }, status: :unprocessable_entity
+      end
+
+      def destroy
+        @display_ad = DisplayAd.find(params[:id])
+        result = @display_ad.update(published: false)
+        render json: @display_ad, status: (result ? :ok : :unprocessable_entity)
```

Comment:
Should we have a rescue block here too?

Reply:
I think it should not be necessary -- the other actions need to rescue from not matching `enum`'s expected inputs, but this endpoint isn't handling that kind of input. (We have `rescue_from` already for `NotFound`.)